internal/pproflogging/pproflogging.go

```diff
@@ -237,30 +284,31 @@ func StartProfileBuffers(ctx context.Context) {
 }
 
 // DumpPem dump a PEM version of the byte slice, bs, into writer, wrt.
-func DumpPem(bs []byte, types string, wrt *os.File) error {
+func DumpPem(ctx context.Context, bs []byte, types string, wrt Writer) error {
```

Comment:
Context can be used to cancel PEM dump.   Unfortunately there's no context aware output in Go :/ - but its a good start.